src/CommandingOfficers/Commander.java

```diff
@@ -132,6 +132,8 @@ public GameEventQueue revertActiveAbility(MapMaster map)
     {
       events.add(new CommanderAbilityRevertEvent(myActiveAbility));
       events.addAll(myActiveAbility.getRevertEvents(map));
```

Comment:
Do we not activate these events before the user gets control? Or is it just that the turn technically starts before the events fire, and by then the player has already gained the vision?

Reply:
The debuff needs to reset before the player's fog vision is reset for the turn (especially for the DoR case), which happens in `Army.initTurn()`.
I guess you could argue that resetting the vision/income/etc should also go in an event that ends up on the end of the "start turn" queue.

Reply:
I could argue that. It might even be cleaner really (not a super fan of moving the contents of `CommanderAbilityRevertEvent` somewhere else), though I don't think of a vision reset as a "game event" per se.

Another possible approach: have `TurnInitEvent.performEvent()` reset vision for the active army and [move its creation to the end of the event queue in `GameInstance.turn()`](https://github.com/ThislsAUsername/DefendPeace/blob/master/src/Engine/GameInstance.java#L276)? I.e, from line 276 down to 291. That seems to fix it on master without modifying Commander, Army or the ability revert event.

Reply:
Hmmmhmmhmm.

I think what I might like to do is:
Get rid of Army/CO `initTurn()`
`Unit.initTurn()` loses its event stuff.
Add a new Army/CO hook for generating new-turn events that doesn't modify state.
If COs need to meddle with their internal state on turn init, `receiveTurnInitEvent()` exists.
`TurnInitEvent` calls `Unit.initTurn()` and does fog reset stuff interactions.

So then `GameInstance.turn()` looks like so:
Queue `ModifyFundsEvent` for income (just so it's first)
Do weather stuff
Queue CO/unit init events
Queue `TurnInitEvent`

This also neatly solves the fog problem by making sure the CO's power revert events go through before the `TurnInitEvent`.

There is one fly in this ointment, and her name is Ave.
Her `initTurn()` both modifies internal state (the snow map) and generates events (applying the snow map).
I guess I could just leave all that in the event-generation hook since it'll only get called once per turn anyway, but it makes me sad.
...or I can just have all her logic go into `receiveTurnInitEvent()`, since listeners are allowed to inject events. Huh.

Thoughts?

Reply:
I ended up doing something a bit different. Plans, enemy, all that.
I had to tweak the tests a bit because now units depend on the `GameEventListener`-returns-events behavior.
But like, it's bad that we weren't accounting for that before anyway, so.

I had to add a separate `ResetVisionEvent` because otherwise you'd see the buffed vision *during the `TurnInitEvent` animation*, since `perform()` happens at the end of that.
But I think I like this design a bit better than I had before.

Reply:
Agree, this does clean things up a fair bit.